### PR TITLE
214 get tx list

### LINKED ISSUE #214: Support GET /transactions (list)

### PR DESCRIPTION
**Pull request by sbellem**
Resolves #214.

### CODE REVIEW

File: bigchaindb_driver/driver.py [hunk 1]
```diff
@@ -193,6 +193,31 @@ def fulfill(transaction, private_keys):
         """
         return fulfill_transaction(transaction, private_keys=private_keys)
 
+    def get(self, *, asset_id, operation=None, headers=None):
+        """Get a list of transactions that have the given asset id and or
+        operation.
```
Comment (sohkai) [inline bigchaindb_driver/driver.py]: > and or

Not exactly true, since the asset id is mandatory. I think I'd reword this to be something like:

> Get a list of transactions that are related to the given asset id (and optionally filtered by their operation).

File: bigchaindb_driver/driver.py [hunk 2]
```diff
@@ -193,6 +193,31 @@ def fulfill(transaction, private_keys):
...
+            asset_id (str): Id of the asset. Please note that the id of
+                an asset in BigchainDB is actually the id of the
+                transaction which created the asset. In other words,
+                when querying for an asset id with the operation set to
+                ``'CREATE'``, only one transaction should be expected.
+                This transaction will be the transaction in which the
+                asset was created, and the transaction id will be equal
+                to the given asset id.
+            operation (str): The type of operation the transaction
+                should be. eITHER ``'CREATE'`` OR ``'TRANSFER'``.
```
Comment (sohkai) [inline bigchaindb_driver/driver.py]: > **E**ITHER

File: tests/conftest.py [hunk 3]
```diff
@@ -132,6 +132,22 @@ def dimi_pubkey(dimi_keypair):
 
 
 @fixture
+def ewy_keypair():
```
Comment (sohkai) [inline tests/conftest.py]: Was this... supposed to be Eve or Evey... 😛

File: tests/conftest.py [hunk 4]
```diff
@@ -346,6 +362,58 @@ def persisted_transfer_carol_car_to_dimi(carol_keypair, dimi_pubkey, bdb_node,
 
 
 @fixture
+def persisted_transfer_dimi_car_to_ewy(dimi_keypair, ewy_pubkey, bdb_node,
+                                       persisted_transfer_carol_car_to_dimi):
+    # FIXME The sleep, or some other approach is required to wait for the
+    # transaction to be available as some processing is being done by the
+    # server.
+    sleep(1.5)
```
Comment (sohkai) [inline tests/conftest.py]: Maybe these sleeps should happen at the end of each fixture rather than requiring the user of the fixture to add them manually?

File: tests/test_driver.py [hunk 5]
```diff
@@ -109,6 +111,45 @@ def test_send(self, driver, alice_privkey, unsigned_transaction):
...
+            driver.transactions.get()
+        assert exc.value.args == (
+            "get() missing 1 required keyword-only argument: 'asset_id'",)
+
+    @mark.parametrize('query_params', (
+        {}, {'operation': 'CREATE'}, {'operation': 'TRANSFER'}
+    ))
+    def test_get_empty(self, driver, query_params):
+        response = driver.transactions.get(
+            asset_id=sha3_256(''.encode()).hexdigest())
```
Comment (sohkai) [inline tests/test_driver.py]: Perhaps `sha3_256(''.encode())` is a bit opaque as to getting the tx id; I'd put this in a fixture or maybe use one of the pre-generated tx dicts.

File: tests/conftest.py (hunk 3 above)
Comment (sbellem) [inline tests/conftest.py]: ha ha, I did some research to get this one ...

Comment (sbellem) [inline tests/conftest.py]: she is going to be a jedi in an upcoming star wars movie

File: tests/conftest.py (hunk 4 above)
Comment (sbellem) [inline tests/conftest.py]: see https://github.com/bigchaindb/bigchaindb/issues/1123

File: tests/test_driver.py (hunk 5 above)
Comment (sbellem) [inline tests/test_driver.py]: Well, ok, this is so to fit the requirement set by https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/web/views/parameters.py#L4 / https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/web/views/transactions.py#L59

Comment (sbellem) [inline tests/test_driver.py]: how simply 

```python
        response = driver.transactions.get(asset_id='a' * 64)
```

File: bigchaindb_driver/driver.py [hunk 6]
```diff
@@ -194,22 +194,38 @@ def fulfill(transaction, private_keys):
...
 
+        Note:
+            Please note that the id of an asset in BigchainDB is
+            actually the id of the transaction which created the asset.
+            In other words, when querying for an asset id with the
+            operation set to ``'CREATE'``, only one transaction should
+            be expected. This transaction will be the transaction in
+            which the asset was created, and the transaction id will be
+            equal to the given asset id. Hence, the following calls to
+            :meth:`~.retrieve` and :meth:`.get` should return the same
```
Comment (sohkai) [inline bigchaindb_driver/driver.py]: > :meth:\`**~**.get\` (should there be an **~**?)

File: bigchaindb_driver/driver.py [hunk 7]
```diff
@@ -194,22 +194,38 @@ def fulfill(transaction, private_keys):
...
-                transaction which created the asset. In other words,
-                when querying for an asset id with the operation set to
-                ``'CREATE'``, only one transaction should be expected.
-                This transaction will be the transaction in which the
-                asset was created, and the transaction id will be equal
-                to the given asset id.
+            asset_id (str): Id of the asset.
             operation (str): The type of operation the transaction
-                should be. eITHER ``'CREATE'`` OR ``'TRANSFER'``.
+                should be. either ``'CREATE'`` or ``'TRANSFER'``.
```
Comment (sohkai) [inline bigchaindb_driver/driver.py]: > **E**ither

File: bigchaindb_driver/driver.py [hunk 8]
```diff
@@ -194,22 +194,38 @@ def fulfill(transaction, private_keys):
         return fulfill_transaction(transaction, private_keys=private_keys)
 
     def get(self, *, asset_id, operation=None, headers=None):
-        """Get a list of transactions that have the given asset id and or
-        operation.
+        """Given an asset id, get its list of transactions (and
+        optionally filter by the operation type (``'CREATE'`` or
```
Comment (sohkai) [inline bigchaindb_driver/driver.py]: I might rephrase the clause inside the brackets to something like

> (and optionally filter for only ``'CREATE'`` or ``'TRANSFER'`` transactions)

to avoid the nested brackets (not really a thing ~~(really)~~).
